src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaTeachingAssignmentService.java

```diff
@@ -22,9 +24,17 @@
 
 	@Inject ScheduleService scheduleService;
 	@Inject InstructorService instructorService;
+	@Inject InstructorTypeService instructorTypeService;
 
 	@Override
 	public TeachingAssignment save(TeachingAssignment teachingAssignment) {
+		// Attempt to automatically fill in the instructorType of the instructor (based on current userRoles). May leave it null if nothing relevant is found.
+		if (teachingAssignment.getInstructor() != null && teachingAssignment.getInstructorType() == null) {
```

Comment:
Should this logic be here? Shouldn't save() be a bit dumb and whatever feeds into save() handle the problem of the instructorType? Presumably other pieces of code will call save() that won't need the instructorType logic to run given that it's a re-usable component.

Reply:
Currently, in every context that calls teachingAssignmentService.save(), this 'addInstructorType' logic is relevant. I thought about writing a separate method that would call save, but that would
- Add unnecessary structure (nothing would call save() directly currently)
- Give me the opportunity to accidentally call save in the future and accidentally not attach instructorTypes.

Or perhaps I could put it in a separate method, and have save call it, to make it clear that there is a separate bit of logic occurring? What do you think?

Reply:
Separate them out. Another developer is never going to guess that save() does something like that, especially given its a generic function name.

Reply:
If having an instructorType really is important, throw an exception here to give you an opportunity to fix the real underlying problem in whatever piece of code needs to set it.